app/src/main/java/org/simple/clinic/contactpatient/ContactPatientBottomSheet.kt

```diff
@@ -162,6 +162,8 @@ class ContactPatientBottomSheet : BaseBottomSheet<
       val selectedDate = result.result as SelectedDate
       val event = ManualDateSelected(selectedDate = selectedDate.date, currentDate = LocalDate.now(userClock))
       hotEvents.onNext(event)
+    } else if (requestType == RemoveOverdueAppointmentResult && result is Succeeded) {
+      router.pop()
```

Comment:
If a single screen can navigate to multiple screens for results, then the individual request types should be a `sealed class` type so that the result handling is enforced. Since this screen has at least two result types, this should be followed here.

Reply:
Yeah, good point.

Reply:
But in this scenario, we cannot make it as a sealed class because we are getting the request type for date picker from the key factory. So `RemoveOverdueAppointmentResult` is the only request type in this screen.

Reply:
Ah, I see. We could still lift it out and use the `when` clause, right? It won't be exhaustive, but might still be easier to read.

Reply:
Yep, I have done that.